plutus-core/src/Language/PlutusCore/Builtins.hs

```diff
@@ -126,10 +118,10 @@ nonZeroArg _ _ 0 = EvaluationFailure
 nonZeroArg f x y = EvaluationSuccess $ f x y
 
 defBuiltinsRuntime :: HasConstantIn DefaultUni term => BuiltinsRuntime DefaultFun term
-defBuiltinsRuntime = toBuiltinsRuntime defDefaultFunDyn defaultCostModel
+defBuiltinsRuntime = toBuiltinsRuntime () defaultCostModel
 
 instance (GShow uni, GEq uni, DefaultUni <: uni) => ToBuiltinMeaning uni DefaultFun where
-    type DynamicPart uni DefaultFun = DefaultFunDyn
+    type DynamicPart uni DefaultFun = ()
```

Comment:
Now that `trace` does not need to be a "dynamic" builtin, we don't need the whole static vs dynamic distinction either. I'm going to remove the `DynamicPart` functionality, but in a subsequence PR, so that it's easy to restore it if we evee need it again.